src/WebJobs.Extensions.Twilio/Config/TwilioSmsConfiguration.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Extensions.Bindings;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Newtonsoft.Json.Linq;
+using Twilio.Clients;
+using Twilio.Rest.Api.V2010.Account;
+using Twilio.Types;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Twilio
+{
+    public class TwilioSmsConfiguration : IExtensionConfigProvider
+    {
+        internal const string AzureWebJobsTwilioAccountSidKeyName = "AzureWebJobsTwilioAccountSid";
+        internal const string AzureWebJobsTwilioAccountAuthTokenName = "AzureWebJobsTwilioAuthToken";
+
+        private readonly ConcurrentDictionary<Tuple<string, string>, TwilioRestClient> _twilioClientCache = new ConcurrentDictionary<Tuple<string, string>, TwilioRestClient>();
+
+        private string _defaultAccountSid;
+        private string _defaultAuthToken;
+
+        public string AccountSid { get; set; }
+
+        public string AuthToken { get; set; }
+
+        public string Body { get; set; }
+
+        public string From { get; set; }
+
+        public string To { get; set; }
+
+        public void Initialize(ExtensionConfigContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            INameResolver nameResolver = context.Config.GetService<INameResolver>();
+
+            _defaultAccountSid = nameResolver.Resolve(AzureWebJobsTwilioAccountSidKeyName);
```

Comment:
instead of using this flow, can we use `[AppSetting(Default="AzureWebJobsTwilioAccountSid")]` instead?